### PR TITLE
Making doc extraction more reliable

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Making doc extraction more reliable by updating the docconv library and
wrapping the extraction process in a panic recovery just in case something
could happen.

#### Ticket Link
[MM-36594](https://mattermost.atlassian.net/browse/MM-36594)

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1:

**Review by ashishbhate — APPROVED** [review]

File: services/docextractor/documents.go [hunk 1]
```diff
@@ -31,7 +31,14 @@ func (de *documentExtractor) Match(filename string) bool {
 	return ok
 }
 
-func (de *documentExtractor) Extract(filename string, r io.ReadSeeker) (string, error) {
+func (de *documentExtractor) Extract(filename string, r io.ReadSeeker) (out string, outErr error) {
+	defer func() {
+		if r := recover(); r != nil {
+			out = ""
+			outErr = errors.New("error extracting pdf text")
```
Comment (ashishbhate) [inline services/docextractor/documents.go:38]: Here it can be any file, not just pdfs?

Comment (jespino) [reply services/docextractor/documents.go:38]: Good point, can be any file.

Comment (jespino) [reply services/docextractor/documents.go]: Done, thanks!

**Comment by jespino** [comment]
@metanerd @cpanato Do you know why is this failing? I don't think is related to my PR. Looks like is not able to get branch of the PR. `error: pathspec 'pull/17842' did not match any file(s) known to git`

**Comment by cpanato** [comment]
> @metanerd @cpanato Do you know why is this failing? I don't think is related to my PR. Looks like is not able to get branch of the PR. `error: pathspec 'pull/17842' did not match any file(s) known to git`

looks like was due to some outage issues in circleci yesterday

**Review by ogi-m — APPROVED** [review]
LGTM
- uploaded doc and docx files
- uploaded docx file with doc extension
- extraction seems to be working fine, was able to search the files by content

**Pull request merged**